### DISCUSSION TITLE
Reference frame for stress output under finite strain?

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#24040] [Q&A Modules: Solid mechanics] [answered]
For a finite strain calculation - what reference frame are the output stress component values in (exodus file or postprocessors)? I'm curious how this compares to Abaqus, which (by default) reports stress in the reference frame it was computed in, which could be the global or an intermediate frame depending on if a specific material orientation is used. Of course Abaqus uses different kinematics (Hughes-Winget rather than Rashid) so maybe that has something to do with it. Is there a difference when AbaqusUMATStress is used? I can follow the logic in that code but not sure if the stress gets rotated elsewhere.

**Comment by jiangwen84** [comment, marked as answer]
If you are using Rashid (incremental) formulation in MOOSE, then the output is Cauchy stress, i.e., respect to deformed configuration and after rotation.

**Reply by jessecarterMOOSE** [reply]
Thanks. So in terms of code, if I'm using [ComputeFiniteStrainElasticStress](https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/src/materials/ComputeFiniteStrainElasticStress.C) for example, when the stress is rotated at the end of the `computeQpStress`, that is the stress that gets reported? Or does stress get rotated elsewhere?

**Reply by jiangwen84** [reply]
Correct. The lines 85-87 in that class rotate the stress from intermediate state (without rotation) to the final rotated state. After that, stress is finalized, and then be outputted.

**Reply by jessecarterMOOSE** [reply]
If I'm reading that code correctly, the intermediate state is the previous rotated state (*old* total rotation)? I think that is consistent with what is being done in `AbaqusUMATStress` where the stress is rotated to the current configuration at the end of the method, which will be will be used to calculate stress on the next step (stress is stateful so it is `stress_old`). Is that consistent throughout the different stress calculators in MOOSE?

Also curious what it means for stress to be finalized?

**Reply by jiangwen84** [reply]
When I say "finalized", I just mean it won't be modified anymore. 

The intermediate state is the "previous rotated state + new incremental stretch", and the rotation matrix that applies to the intermediate state is the incremental rotation. I believe this is consistent through all. Are you using plasticity, and damage? If so, maybe you want to check the code ComputeFiniteStrainElasticStress.

**Reply by jessecarterMOOSE** [reply]
> Are you using plasticity, and damage? If so, maybe you want to check the code ComputeFiniteStrainElasticStress.

Eventually, but just trying to start simple. Did you mean a different class to check out?

**Reply by jiangwen84** [reply]
My bad. ComputeMultipleInelasticStress

**Reply by jessecarterMOOSE** [reply]
And is it this stress value that gets used in the stress divergence calculation or is it rotated along the way?

**Reply by jiangwen84** [reply]
Correct, this is the stress that is used in the stress divergence calculation, to compute residual and Jacobian for a new nonlinear iteration. After solving the nonlinear step, you will get new displacements/strain, and then repeat the same thing to update stress. It will repeat until nonlinear iteration converges.

**Reply by jessecarterMOOSE** [reply]
When should the `use_finite_deform_jacobian` parameter be set? Doesn't look like it gets set by the Tensor Mechanics Master Action when doing finite strain. I think it's doing a stress rotation in that case.

**Reply by jiangwen84** [reply]
```use_finite_deform_jacobian``` is used to improve the Jacobian calculation by adding a few chain rule derivatives. The improvement is very limited, so it is not included in the action. It should not affect the results anyway. If you are using AD, you definitely do not need this option on.
